frontend/javascripts/oxalis/api/api_latest.js

```diff
@@ -1219,20 +1227,70 @@ class DataApi {
     return bucket;
   }
 
-  async getDataFor2DBoundingBox(layerName: string, bbox: BoundingBoxType) {
-    const bucketAddresses = this.getBucketAddressesInCuboid(bbox);
+  async getDataFor2DBoundingBox(
+    layerName: string,
+    bbox: BoundingBoxType,
+    _zoomStep: ?number = null,
+  ) {
+    const layer = getLayerByName(Store.getState().dataset, layerName);
+
+    let zoomStep;
+    if (_zoomStep != null) {
+      zoomStep = _zoomStep;
+    } else {
+      const resolutionInfo = getResolutionInfo(layer.resolutions);
+      zoomStep = resolutionInfo.getClosestExistingIndex(0);
+    }
+
+    const { resolutions } = layer;
+    const bucketAddresses = this.getBucketAddressesInCuboid(bbox, resolutions, zoomStep);
+    if (bucketAddresses.length > 15000) {
+      console.warn(
+        "More than 15000 buckets need to be requested for the given bounding box. Consider passing a smaller bounding box or using another resolution.",
+      );
+    }
     const buckets = await Promise.all(
       bucketAddresses.map(addr => this.getLoadedBucket(layerName, addr)),
     );
     const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
-    return this.cutOutCuboid(buckets, bbox, elementClass);
+    return this.cutOutCuboid(buckets, bbox, elementClass, resolutions, zoomStep);
+  }
+
+  async getDataForViewport(viewport: OrthoView, layerName: string) {
```

Comment:
I think, naming-wise `getViewportData` would be a bit better :) Sorry, I think, the name was even my suggestion, but it just crossed my mind..